src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -167,61 +172,82 @@ export default class UBAFeeSpokeCalculator {
+   * @param flowRange The range of the flow to simulate the event for. Defaults to undefined to simulate the event for the entire flow.
+   * @returns The event fee for the spoke and the given symbol and a given flow type
+   */
+  protected getEventFee(amount: BigNumber, flowType: "inflow" | "outflow", flowRange?: UBAFlowRange): FlowFee {
+    // The rough psuedocode for this function is as follows:
+    // We'll need two inflow/outflow curves
+    // We need to determine which flow curve to use based on the flow type
+    // Compute first balancing fee <- f(x, x+amnt)
+    // Compute second balance fee (oportunity cost) <- g(x+amnt, x)
+    // Incentive Fee (LP Fee Component): first balancing fee - second balance fee
+    // Return (LP Fee + Balancing Fee)
+    // #############################################
+
+    // We'll need to now compute the concept of the running balance of the spoke
+    const { runningBalance } = this.calculateHistoricalRunningBalance(flowRange?.startIndex, flowRange?.endIndex);
+
+    // We first need to resolve the inflow/outflow curves for the deposit and refund spoke
+    const flowCurves = this.config.getBalancingFeeTuples(this.chainId);
+
+    // We'll need to determine which flow curve to use based on the flow type
+    const mainFlowCurve = flowCurves[flowType];
+    const reverseFlowCurve = flowCurves[flowType === "inflow" ? "outflow" : "inflow"];
+
+    // We'll need to figure out the starting runningBalance of the spoke depending on the flow type
+    const mainIntegrandStart = flowType === "inflow" ? runningBalance : runningBalance.add(amount);
+    const reverseIntegrandStart = flowType === "inflow" ? runningBalance.add(amount) : runningBalance;
+
+    // For convenience, let's resolve a structure that makes resolving functions easy to read
+    const functionLookup = {
+      inflow: getDepositBalancingFee,
+      outflow: getRefundBalancingFee,
+    };
+
+    // We'll need to resolve the fee function for the main and reverse integrand
+    const mainIntegrandFeeFunction = functionLookup[flowType];
+    const reverseIntegrandFeeFunction = functionLookup[flowType === "inflow" ? "outflow" : "inflow"];
+
+    // Next, we'll need to compute the first balancing fee from the running balance of the spoke
+    // to the running balance of the spoke + the amount
+    const mainFee = mainIntegrandFeeFunction(mainFlowCurve, mainIntegrandStart, amount);
+
+    // Next, we'll need to compute the reverse balancing fee. This is the opportunity cost of
+    // the LP fee holders to compute the opposite flow and to essentially reverse the flow
+    const reverseFee = reverseIntegrandFeeFunction(reverseFlowCurve, reverseIntegrandStart, amount);
+
+    // We can now compute the LP fee component of the fee. This is also considered the incentive fee
+    const lpFee = mainFee.sub(reverseFee);
+
+    // We can now return the fee
+    return {
+      lpFee,
+      balancingFee: mainFee.sub(lpFee),
```

Comment:
As above, would we consider `incentive` rather than `balancingFee` ?

(I'm less thrilled about `incentiveFee` because it seems like the conjunction of two opposites)